src/test/java/io/cdap/plugin/ariba/source/AribaServicesTest.java

```diff
@@ -0,0 +1,746 @@
+    "          \"scale\": null,\n" +
+    "          \"size\": 100,\n" +
+    "          \"allowedValues\": null\n" +
+    "        },\n" +
+    "        \"UserId\": {\n" +
+    "          \"title\": \"Owner.UserId\",\n" +
+    "          \"type\": [\n" +
+    "            \"string\",\n" +
+    "            \"null\"\n" +
+    "          ],\n" +
+    "          \"precision\": null,\n" +
+    "          \"scale\": null,\n" +
+    "          \"size\": 50,\n" +
+    "          \"allowedValues\": null\n" +
+    "        }\n" +
+    "      }\n" +
+    "    },\n" +
+    "     \"Organization\": {\n" +
+    "      \"type\": [\n" +
+    "        \"array\",\n" +
+    "        \"null\"\n" +
+    "      ],\n" +
+    "      \"items\": [\n" +
+    "        {\n" +
+    "          \"type\": [\n" +
+    "            \"object\",\n" +
+    "            \"null\"\n" +
+    "          ],\n" +
+    "          \"properties\": {}\n" +
+    "        },\n" +
+    "         {\n" +
+    "          \"type\": [\n" +
+    "            \"array\",\n" +
+    "            \"null\"\n" +
+    "          ],\n" +
+    "          \n" +
+    "  \"items\": [\n" +
+    "        {\n" +
+    "          \"type\": [\n" +
+    "            \"object\",\n" +
+    "            \"null\"\n" +
+    "          ],\n" +
+    "          \"properties\": {}\n" +
+    "        },\n" +
+    "        {\n" +
+    "          \"type\": [\n" +
+    "            \"array\",\n" +
+    "            \"null\"\n" +
+    "          ],\n" +
+    "          \"properties\": {}\n" +
+    "        }\n" +
+    "      ]\n" +
+    "        }\n" +
+    "        \n" +
+    "      ]\n" +
+    "    }\n" +
+    "  }\n" +
+    "}";
+  private AribaServices aribaServices;
+  private AribaPluginConfig pluginConfig;
+  private MockPipelineConfigurer pipelineConfigurer;
+
+  @Before
+  public void setUp() {
+    pipelineConfigurer = new MockPipelineConfigurer(null);
+    pluginConfig = new AribaPluginConfig("unit-test-ref-name", "https://openapi.au.cloud.ariba.com",
+                                         "prod", "CloudsufiDSAPP-T",
+                                         "SourcingProjectFactSystemView", "08ee0299-4849-42a4-8464-3abed75fc74e",
+                                         "c3B5wvrEsjKucFGlGkkhKSWUDqDRGE2Wds", "xryi0757SU8pEyk7ePc7grc7vgDXdz8O",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+  }
+
+  @Test
+  public void testGetAccessToken() throws AribaException, IOException {
+    aribaServices = new AribaServices(pluginConfig);
+
+    String tokenUrl = String.format("https://%s", "https://api.au.cloud.ariba.com/v2/oauth/token");
+    HttpUrl.Builder builder = Objects.requireNonNull(HttpUrl.parse(tokenUrl))
+      .newBuilder().addPathSegments("TOKEN_PATH");
+
+    InputStream inputStream = new ByteArrayInputStream(jsonNode.getBytes(StandardCharsets.UTF_8));
+    new Expectations(AribaServices.class) {
+      {
+        aribaServices.generateTokenURL();
+        result = any;
```

Comment:
what are minTimes, result? where are they used?

Reply:
minTimes = number of time new Expectation block will run.
result = mocked result when the above written method is being called.
they will be used in the entire flow whenever the specified method is being called.